public/javascript/_common/services/providers/thirdPartyProviders/soundcloud/player/soundcloudPlayer.service.js

```diff
@@ -0,0 +1,143 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('ammo.soundcloud.player.service', ['ammo.event.service'])
+    .factory('soundcloudPlayer', soundcloudPlayerService);
+
+  function soundcloudPlayerService($q, $timeout, event) {
+    var playerLoaded = false,
+        service,
+        soundcloud;
+
+    service = {
+      mute: mute,
+      pause: pause,
+      play: play,
+      resume: resume,
+      seekTo: seekTo,
+      setVolume: setVolume,
+      unmute: unmute
+    };
+
+    return service;
+
+    ////////////
+    /**
+     * Loads a song to the soundcloud player and then plays it
+     * @params {object} song Song to play
+     **/
+    function play(song) {
+      loadPlayer()
+        .then(function successLoadPlayer() {
+          return setStream(song);
+        })
+        .then(function successSetStream() {
+          soundcloud.play();
+        })
+        .catch(function errorLoadPlayer() {
+          event.publish('error', 'Failed to load soundcloud Player');
+        });
+    }
+
+    function pause() {
+      soundcloud.pause();
+    }
+
+    function resume() {
+      soundcloud.resume();
+    }
+
+    function mute() {
+      soundcloud.mute();
+    }
+
+    function unmute() {
+      soundcloud.unmute();
+    }
+
+    function seekTo(second) {
+      soundcloud.setPosition(second * 1000);
+    }
+
+    function setVolume(value) {
+      soundcloud.setVolume(value);
+    }
+
+
+    ///////
+    function setStream(song) {
+      var deferred = $q.defer(),
+          newEvent = { service: 'soundcloud' };
+
+      SC.stream('/tracks/' + song.serviceId, {
+        onfinish: function() {
+          event.publish('ended', newEvent);
+        },
+        onplay: function() {
+          event.publish('playing', newEvent);
+        },
+        onresume: function() {
+          event.publish('playing', newEvent);
+        },
+        onpause: function() {
+          event.publish('paused', newEvent);
+        },
+        onbufferchange: function() {
+          if (this.isBuffering) {
+            event.publish('buffering', newEvent);
+          } else {
+            event.publish('playing', newEvent);
+          }
+        }
+      }, function(scObject) {
+        soundcloud = scObject;
+        deferred.resolve();
```

Comment:
:tropical_drink: resolve scObject inside the resolve, and set soundcloud inside the then statement. i thin it makes the flow a little more readable